### ISSUE TITLE
Whether `otel` related components are required for the basic function of the "config-sync"

### ISSUE THREAD

**Issue by Shuanglu** [GoogleContainerTools/kpt-config-sync#863]
## WHAT
I noticed that in the [manifest](https://github.com/GoogleContainerTools/kpt-config-sync/releases/tag/v1.15.0-rc.6) there are some components like namespace 'config-management-monitoring'. It looks like they are only used for metrics collection. If I only want the basic feature of the `config-sync` like reconciling namespace/rolebinding, is it ok to remove them? 

- namespaces/config-management-monitoring
- namespaces/config-management-system/configMap/otel-agent
- namespaces/resource-group-system/configMap/resource-group-otel-agent

**Comment by mikebz** [comment]
I don't think we test that scenario (there is enough permutations we are doing), but it could be that it's not needed @tiffanny29631 can chime in here.

**Comment by karlkfi** [comment]
The otel-agent sidecar is always installed in the config sync deployments. So if the otel-collector is not present on the cluster, those sidecars will error. And if you manually remove the sidecars, then the main app binaries would probably error and crash or log errors endlessly.

If this is a scenario you want to see, we could consider it a feature request though. FWIW, we have been considering making the metric export more easily configurable. It's possible we conceivably add an option to disable metrics completely, but it's not really something we would recommend doing in production.

Do you have a specific use case where you want to disable the metrics?

**Comment by Shuanglu** [comment]
> The otel-agent sidecar is always installed in the config sync deployments. So if the otel-collector is not present on the cluster, those sidecars will error. And if you manually remove the sidecars, then the main app binaries would probably error and crash or log errors endlessly.
> 
> If this is a scenario you want to see, we could consider it a feature request though. FWIW, we have been considering making the metric export more easily configurable. It's possible we conceivably add an option to disable metrics completely, but it's not really something we would recommend doing in production.
> 
> Do you have a specific use case where you want to disable the metrics?

Thanks. It's just that we don't care much about the metrics itself. At least for now, the config-sync itself runs well and we don't find it's necessary for us to monitor the metrics of it. We only want it to run and sync the resources in the repo to our GKE.

**Comment by tiffanny29631** [comment]
We could either enable the metric exporting only when authentication is configured, or use an additional API field to configure the metric pipeline. 

Suggest something similar to the following in the Rsync spec:
```
spec:
  metrics:
    - exporter: prometheus
    - exporter: googlecloud
    - exporter: custom
```

Only configured pipelines get enabled.

**Comment by tiffanny29631** [comment]
We suggest adding an env var for reconciler-manager to disable the otel-agent sidecar for the Rootsync and Reposync Reconcilers to reduce resource comsumption, and to deregister the metric export from resource-group-controller and reconciler-manager. With this change, when metrics are disabled, users should be safe to remove the config-management-monitoring namespace and underlying resources.

In the MVP design we don't introduce API change to minimize the dev cycle and to have the new behavior tested before API change is made.

**Comment by Shuanglu** [comment]
> adding an env var for reconciler-manager

@tiffanny29631 thanks a lot for the follow up. Would you mind elaborating more on this one? e.g. which env var can be used to disable it
